src/Catalyst.Core.Modules.Dfs/DagNode.cs

```diff
@@ -0,0 +1,355 @@
+                    stream.WriteSomeBytes(msg);
+                }
+            }
+
+            if (DataBytes.Length > 0)
+            {
+                stream.WriteTag(1, WireFormat.WireType.LengthDelimited);
+                stream.WriteLength(DataBytes.Length);
+                stream.WriteSomeBytes(DataBytes);
+            }
+        }
+
+        void Read(Stream stream)
+        {
+            using (var cis = new CodedInputStream(stream, true))
+            {
+                Read(cis);
+            }
+        }
+
+        void Read(CodedInputStream stream)
+        {
+            var links = new List<DagLink>();
+            bool done = false;
+
+            while (!stream.IsAtEnd && !done)
+            {
+                var tag = stream.ReadTag();
+                switch (WireFormat.GetTagFieldNumber(tag))
+                {
+                    case 1:
+                        DataBytes = stream.ReadSomeBytes(stream.ReadLength());
+                        done = true;
+                        break;
+                    case 2:
+                        using (var ms = new MemoryStream(stream.ReadSomeBytes(stream.ReadLength())))
+                        {
+                            links.Add(new DagLink(ms));
+                        }
+
+                        break;
+                    default:
+                        throw new InvalidDataException("Unknown field number");
+                }
+            }
+
+            if (DataBytes == null)
+                DataBytes = new byte[0];
+            Links = links.ToArray();
+        }
+
+        /// <summary>
+        ///   Returns the IPFS binary representation as a byte array.
+        /// </summary>
+        /// <returns>
+        ///   A byte array.
+        /// </returns>
+        public byte[] ToArray()
+        {
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                return ms.ToArray();
+            }
+        }
+
+        void ComputeHash()
+        {
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                size = ms.Position;
+                ms.Position = 0;
+                id = MultiHash.ComputeHash(ms, hashAlgorithm);
+            }
+        }
+
+        void ComputeSize()
+        {
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                size = ms.Position;
+            }
+        }
+        
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var that = obj as DagNode;
+            return that != null && this.Id == that.Id;
+        }
+
+        /// <inheritdoc />
+        public bool Equals(DagNode that) { return that != null && this.Id == that.Id; }
+
+        /// <summary>
+        ///  TODO
+        /// </summary>
+        public static bool operator ==(DagNode a, DagNode b)
+        {
+            if (object.ReferenceEquals(a, b)) return true;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)